tools/engine_tool/lib/src/commands/cleanup_command.dart

```diff
@@ -0,0 +1,165 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:meta/meta.dart';
+import 'package:path/path.dart' as p;
+
+import '../logger.dart';
+import 'command.dart';
+import 'flags.dart';
+
+/// The `cleanup` command for removing unnecessary on-disk artifacts.
+final class CleanupCommand extends CommandBase {
+  /// Constructs the `cleanup` command.
+  CleanupCommand({
+    required super.environment,
+    super.help = false,
+    super.usageLineLength,
+  }) {
+    argParser.addFlag(
+      dryRunFlag,
+      abbr: 'd',
+      help: 'Write changes to stdout without modifying the file system.',
+      negatable: false,
+    );
+
+    argParser.addOption(
+      'untouched-since',
+      defaultsTo: () {
+        const thirtyDays = Duration(days: 30);
+        final dateTime = environment.now().subtract(thirtyDays);
+        return _toDateString(dateTime);
+      }(),
+      help: 'What date to consider artifacts old enough to safely remove.',
+      valueHelp: 'YYYY-MM-DD',
+    );
+  }
+
+  @override
+  String get name => 'cleanup';
+
+  @override
+  String get description => 'Removes stale or unnecessary on-disk artifacts.';
+
+  @override
+  List<String> get aliases => const ['gc'];
+
+  static final _dateString = RegExp(r'^(\d{4})-(\d{2})-(\d{2})$');
+
+  @override
+  Future<int> run() async {
+    final dryRun = argResults!.flag('dry-run');
+    final since = () {
+      final yyyyMmDd = argResults!.option('untouched-since')!;
+      final dateMatch = _dateString.matchAsPrefix(yyyyMmDd);
+      if (dateMatch == null) {
+        throw FatalError('Invalid --untouched-since: $yyyyMmDd');
+      }
+      return DateTime(
+        int.parse(dateMatch.group(1)!),
+        int.parse(dateMatch.group(2)!),
+        int.parse(dateMatch.group(3)!),
+      );
+    }();
+
+    // Look at the directories in "out" for ones older than "since".
+    environment.logger.status('Checking ${environment.engine.outDir.path}...');
+    final toDelete = [
+      for (final entity in environment.engine.outDir.listSync())
```

Comment:
we can't do `await for` in the list expansion can we?

Reply:
We can (and do now)!